### ISSUE TITLE
Minimum needs calculator error on census data

### ISSUE THREAD

**Issue by myarjunar** [inasafe/inasafe#3927]
### Problem
Minimum needs calculator error on census from test data.

1.
![error while calculating minimum needs_025](https://cloud.githubusercontent.com/assets/11134669/23243094/49fb18d4-f9af-11e6-8f95-2bf4af0fa699.png)

2. 
```
Traceback (most recent call last):
  File "/home/myarjunar/Work/Kartoza/InaSAFE/inasafe-dev/safe/gui/tools/minimum_needs/needs_calculator_dialog.py", line 205, in accept
  del self.result_layer.keywords
AttributeError: 'NoneType' object has no attribute 'keywords'
```

### Expected Behaviour
Works flawlessly

### Environment
* InaSAFE : 4.0
* QGIS : 2.14.11-Essen Essen
* OS : Ubuntu
* Data used (filenames and paths so as to reproduce, a project is better):
  * Exposure : safe/test/data/exposure/census.geojson

### Proposed Solution
1st problem: handle feature that has invalid geometry or has no geometry
2nd problem: if exception raised, return. don't continue the code.

**Comment by myarjunar** [comment]
turns out that there is a feature in census data that has no geometry. we should handle that.

![qgis 2 14 11-essen_027](https://cloud.githubusercontent.com/assets/11134669/23245204/6bc27f6c-f9bd-11e6-8fd7-fd21f21de123.png)
![census features total 4 filtered 4 selected 4_026](https://cloud.githubusercontent.com/assets/11134669/23245210/76fe23f4-f9bd-11e6-81d8-43b7427c5158.png)

**Comment by myarjunar** [comment]
closed in #3929

**Comment by Gustry** [comment]
Thanks @myarjunar I often forgot this layer has a missing geometry too.
